### PR TITLE
time picker only taking input between given range

### LINKED ISSUE #72: Create Time functions for local flexible gatepass
Create the time function as discussed in the meeting. 
Create the folder structure for local flexible as well and start fresh without copying the old code.

### PR DESCRIPTION
**Pull request by karamarora20**
for issue 72 the files have been created as discussed before

### CODE REVIEW

File: client-side/package-lock.json [hunk 1]
```diff
@@ -22,8 +22,12 @@
         "express": "^4.18.2",
```
Comment (ayushSinha16) [inline client-side/package-lock.json:22]: Why are there so many changes in the package-lock.json file? Have you installed some new package or dependency?

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const[minute,setminute]=useState(new Date().getMinutes());//present minute
+  const[max_time,setmaxTime]=useState([]);//maximum time accepted from backend
+  useEffect(() => {
+    fetch("http://127.0.0.1:4000/gatepass/v2/admin/parameter_config")
+      .then((response) => response.json())
+      .then((text) => {
+        setmaxTime(text[2]["value"].split(':'));
+      });
+  }, []);  
+  console.log(max_time)
```
Comment (ayushSinha16) [inline client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: remove this console.log

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState, useReducer } from "react";
+import ReactDropdown from "react-dropdown";
+import moment from 'moment';
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import "./style.scss";
+
+
+const LFform = (props) => {
+  const[hour,settime]=useState(new Date().getHours());// present time
```
Comment (ayushSinha16) [inline client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: setState should always be camelcase, and the name in the setState should be that of the state you have made. For ex - 
[user, setUser] is correct, but not [user,setCandidate]

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  const[minute,setminute]=useState(new Date().getMinutes());//present minute
```
Comment (ayushSinha16) [inline client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: Camelcase

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  const[max_time,setmaxTime]=useState([]);//maximum time accepted from backend
```
Comment (ayushSinha16) [inline client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: CamelCase and "_" is missing from setState

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  }, []);  
+  console.log(max_time)
+
+  return (
+    <div >
+        <DatePicker
+      // selected={selectedDate}
+      // onChange={handleDateChange}
+      showTimeSelect
+        showTimeSelectOnly
```
Comment (ayushSinha16) [inline client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: indentation

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      timeFormat="HH:mm"
+      timeIntervals={15}
+      minTime={new Date().setHours(hour+2, minute)} // set the minimum time accepted i.e current time +2 hours
```
Comment (ayushSinha16) [inline client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js:28]: Also does this Date() function return the current system date? If yes, then I could change my system date and hence braek the workflow of the code. Do check this out.

**Review by ayushSinha16 — CHANGES_REQUESTED** [review]

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js (hunk 7 above)
Comment (karamarora20) [reply client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js:28]: yes it irrespective of the system time I have verified.

File: client-side/package-lock.json (hunk 1 above)
Comment (karamarora20) [reply client-side/package-lock.json:22]: some new dependancies were installed such react datepicker. Other necessary changes have been committed

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js (hunk 2 above)
Comment (karamarora20) [reply client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: done

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js (hunk 3 above)
Comment (karamarora20) [reply client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: corrected

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js (hunk 4 above)
Comment (karamarora20) [reply client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: corrected

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js (hunk 5 above)
Comment (karamarora20) [reply client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: done

File: client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js (hunk 6 above)
Comment (karamarora20) [reply client-side/src/Components/studentComponent/Gatepasses/LocalFlexible/Form/LFform.js]: corrected

**Review by ayushSinha16 — APPROVED** [review]
ok

**Pull request merged**